modules/cart/src/Form/AddToCartForm.php

```diff
@@ -227,7 +264,7 @@ public function buildEntity(array $form, FormStateInterface $form_state) {
    * @return \Drupal\commerce_store\Entity\StoreInterface
    *   The selected store.
    */
-  protected function selectStore(PurchasableEntityInterface $entity) {
+  public function selectStore(PurchasableEntityInterface $entity) {
     $stores = $entity->getStores();
```

Comment:
also made AddToCartForm::selectStore() public for the convenience of form alters.

Reply:
I don't like this. People should be providing their own add to cart form if they're altering this much.

Reply:
I'm sad that this function is protected now. Current cart form behavior is to show "add to cart" no matter what the availability is. So it can say out of stock, you can click the button and it doesn't add the item to the cart.  I Am using hook_form_alter and the enhanced $form_object->getAvailabilityManager()->check of this PR to change the button to "Unavailable". To do that, I need to construct a $context, which requires the $store.  This seems like a reasonable use of form_alter to me, if the stock AddToCart form isn't going to be availability-aware.